Services/EmailService.cs

```diff
@@ -0,0 +1,139 @@
+﻿using MimeKit;
+using MailKit.Net.Smtp;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using System;
+using System.Threading;
+using System.Collections.Generic;
+using Phytime.Models;
+using Microsoft.EntityFrameworkCore;
+using System.Xml;
+using System.ServiceModel.Syndication;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Phytime.Services
+{
+    public class EmailService : IHostedService, IDisposable
+    {
+        private Timer _timer;
+        private RssSource _rssSource;
+        //private PhytimeContext _context;
+        private readonly IServiceScopeFactory _scopeFactory;
+
+        public EmailService(IServiceScopeFactory scopeFactory)
+        {
+            _rssSource = RssSource.getInstance();
+            _scopeFactory = scopeFactory;
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            _timer = new Timer(CheckUrls, null, TimeSpan.FromSeconds(60), TimeSpan.FromSeconds(30));
+
+            return Task.CompletedTask;
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            _timer?.Change(Timeout.Infinite, 0);
+
+            return Task.CompletedTask;
+        }
+
+        public void Dispose()
+        {
+            _timer?.Dispose();
+        }
+
+        public void CheckUrls(object state)
+        {
+            using (var scope = _scopeFactory.CreateScope())
+            {
+                PhytimeContext _context = scope.ServiceProvider.GetRequiredService<PhytimeContext>();
+                for (int i = 0; i < _rssSource.Urls.Count; i++)
+                {
+                    XmlReader reader = XmlReader.Create(_rssSource.Urls[i]);
+                    SyndicationFeed feed = SyndicationFeed.Load(reader);
+                    reader.Close();
+                    var rssFeed = _context.Feeds.FirstOrDefault(f => f.Url == _rssSource.Urls[i]);
+                    if (rssFeed != null)
+                    {
+                        if (/*true*/rssFeed.ItemsCount != feed.Items.ToList().Count)
+                        {
+                            SendNotifications(_rssSource.Urls[i], _rssSource.Titles[i]);
```

Comment:
first i - rename to rssSourceFeed for example
second, you can pass the only rssSourceFeed to SendNotifications
or SendNotifications should be an another abstraction